### ISSUE TITLE
`electron-store` with `ERB` documentation 📚

### ISSUE THREAD

**Issue by samcodee** [electron-react-boilerplate/electron-react-boilerplate#3022]
If you want to add electron-store, see my [Github Gist](https://gist.github.com/samcodee/d4320006d366a2c47048014644ddc375)

---

If you are a maintainer of this project, please consider adding these to the docs.

**Comment by amilajack** [comment]
https://github.com/electron-react-boilerplate/site/commit/bc533775e75bd4cd422b9f7fcfe1a5c224bc6adf

**Comment by GioLogist** [comment]
Heads up, this isn't working as is. In the case below, nothing is logged to console, because nothing is ever returned from the `window.electron.store.get` function.

```js
// src/renderer/App.tsx
<button
  onClick={() => {
    window.electron.store.set("foo", "bar");
    // or
    console.log(window.electron.store.get("foo"));
  }}
>
  Click Me!
</button>
```

A few solutions are possible. Mainly, you can utilize [sendSync](https://www.electronjs.org/docs/latest/api/ipc-main#sending-messages) in preload.js. Or, you can use the handle/invoke pattern as shown in the [electron-store docs](https://github.com/sindresorhus/electron-store#how-do-i-get-store-values-in-the-renderer-process-when-my-store-was-initialized-in-the-main-process).

**Comment by shahneil** [comment]
@GioLogist 
The fix for the `window.electron.store.get` is below. If someone could update the doc page https://electron-react-boilerplate.js.org/docs/electron-store that would be great.

```
// src/main/main.ts
ipcMain.on('electron-store-get', async (event, property) => {
  event.returnValue = electronStore.get(property);
});
ipcMain.on('electron-store-set', async (_event, property, val) => {
  electronStore.set(property, val);
});```
```

**Comment by amilajack** [comment]
Updated https://electron-react-boilerplate.js.org/docs/electron-store/

**Comment by albanmario** [comment]
@shahneil not working :(

**Comment by GioLogist** [comment]
This won't work because, still, nothing is actually being returned from the `window.electron.store.get` function. Here's what the docs show:

```js
// preload.js
    get(val) {
      ipcRenderer.send("electron-store-get", val);
    },
```

So, when you get a value, it's not actually returning it back to `App.tsx`. And even if it was, `ipcRenderer.send` ([docs](https://www.electronjs.org/docs/latest/api/ipc-renderer#ipcrenderersendchannel-args)) would be returning a promise. Not the actual value. Instead, you want to return `ipcRenderer.sendSync` ([docs](https://www.electronjs.org/docs/latest/api/ipc-renderer#ipcrenderersendsyncchannel-args)).

So, in short, it would be something like this

```js
//main.js
ipcMain.on("electron-store-get", (event, key) => {
  event.returnValue = store.get(key)
});
```

```js
// preload.js
    get(val) {
      return ipcRenderer.sendSync("electron-store-get", val);
    }
```

**Comment by albanmario** [comment]
@GioLogist thanks!!

**Comment by swyxio** [comment]
verified that @GioLogist's solution works in 2023, on electron v25

**Comment by samcodee** [comment]
> This won't work because, still, nothing is actually being returned from the `window.electron.store.get` function. Here's what the docs show:
> 
> ```js
> // preload.js
>     get(val) {
>       ipcRenderer.send("electron-store-get", val);
>     },
> ```
> 
> So, when you get a value, it's not actually returning it back to `App.tsx`. And even if you returned in the function above,`ipcRenderer.send` ([docs](https://www.electronjs.org/docs/latest/api/ipc-renderer#ipcrenderersendchannel-args)) doesn't return anything. Instead, you want to return `ipcRenderer.sendSync` ([docs](https://www.electronjs.org/docs/latest/api/ipc-renderer#ipcrenderersendsyncchannel-args)).
> 
> So, in short, it would be something like this
> 
> ```js
> //main.js
> ipcMain.on("electron-store-get", (event, key) => {
>   event.returnValue = store.get(key)
> });
> ```
> 
> ```js
> // preload.js
>     get(val) {
>       return ipcRenderer.sendSync("electron-store-get", val);
>     }
> ```

It has literally been years since I authored this issue and the gist, I am sorry that my solution was not working. This solution did work a few years ago. It might have stopped working now, due to some changes in the libraries. Anyway, thanks for sharing the new and working solution!
